File: solutions_5706278382862336_0/C++/TouchMyCode/1C_1.cpp
#include<iostream>
#include<fstream>
using std::endl;
unsigned long long GCD(unsigned long long a, unsigned long long b);
int main()
{

	std::ofstream fout;
	std::ifstream fin;
	fin.open("1.txt");
	if (fin.bad()){
		std::cout << "INPUT FILE OPEN ERROR" << std::endl;
		system("PAUSE");
		return 0;
	}
	fout.open("1.out");
	if (fout.bad()){
		std::cout << "OUTPUT FILE OPEN ERROR" << std::endl;
		fin.close();
		system("PAUSE");
		return 0;
	}
	int noQuestion;
	fin >> noQuestion;

	for (int caseNo = 0; caseNo < noQuestion; caseNo++)
	{
		//todo:각 문제 풀기
		bool possible = true;
		unsigned long long v1, v2;
		fin >> v1;
		fin.get();
		fin >> v2;

		unsigned long long v3 = GCD(v1, v2);
		v1 /= v3;
		v2 /= v3;
		v3 = 1;

		while (v2>v3) v3 *= 2;
		if (v2 != v3)possible = false;

		int answer = 1;
		while (answer<50){
			unsigned long long v4 = v2 >> answer;
			if ((v4&v1) != 0)
				break;
			answer++;
		}


		if(possible)
			fout << "Case #" << caseNo+1 << ": " << answer << std::endl;
		else{
			fout << "Case #" << caseNo + 1 << ": " << "impossible" << std::endl;

		}
	}
	fout.close();
	fin.close();
}

unsigned long long GCD(unsigned long long a, unsigned long long b){
	if (a > b){
		long long c = a%b;
		if (c == 0)return b;
		return GCD(b, c);
	}
	else if (a<b){
		long long c = b%a;
		if (c == 0)return a;
		return GCD(a, c);
	}
	else return a;
}

File: solutions_5706278382862336_1/Python/hero777/1c1.py
from fractions import gcd
f = open('a.txt', 'r')
f0 = open('output.txt', 'w')
a0 = [int(x) for x in f.readline().split()]
for index in range (0,a0[0]):
	x,y = [int(x) for x in f.readline().split('/')];
	print(x)
	print(y)
	z = int(gcd(x,y));
	#print(x/z)
	x = int(x/z);
	#print(y/z)
	y = int(y/z);
	f0.write('Case #')
	f0.write(str(index+1))
	f0.write(': ')
	if (x >y):
		f0.write('impossible'+ '\n');
		continue;
	if (x==y):
		f0.write('0' + '\n');
		continue;
	a = bin(x)[2:];
	print(a)
	b = bin(y)[2:];
	print(b)
	if (len(b) == 1):
		f0.write('0'+'\n');
		continue;
	c = int(b[1:],2);
	if (not c == 0):
		f0.write('impossible'+ '\n');
		continue;
	f0.write(str(len(b)-len(a))+'\n');
 
		
		
	
	
	
